src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -843,18 +859,31 @@ static int TopicPartition_init (PyObject *self, PyObject *args,
 	const char *topic;
 	int partition = RD_KAFKA_PARTITION_UA;
 	long long offset = RD_KAFKA_OFFSET_INVALID;
+	const char *metadata;
+	size_t metadata_size = 0;
+	static char *kws_optional[] = { "topic",
+					"partition",
+					"offset",
+					"metadata",
+					"metadata_size",
```

Comment:
I don't think the metadata_size is needed, there's already a length on binary/string objects.